Library/Homebrew/unpack_strategy/directory.rb

```diff
@@ -20,11 +20,29 @@ def self.can_extract?(path)
 
     sig { override.params(unpack_dir: Pathname, basename: Pathname, verbose: T::Boolean).void }
     def extract_to_dir(unpack_dir, basename:, verbose:)
-      path.children.each do |child|
-        system_command! "cp",
-                        args:    ["-pR", (child.directory? && !child.symlink?) ? "#{child}/." : child,
-                                  unpack_dir/child.basename],
-                        verbose:
+      path_children = path.children
+      return if path_children.empty?
+
+      existing = unpack_dir.children
+
+      # We run a few cp attempts in the following order:
+      #
+      # 1. Start with `-al` to create hardlinks rather than copying files if the source and
+      #    target are on the same filesystem. On macOS, this is the only cp option that can
+      #    preserve hardlinks but it is only available since macOS 12.3 (file_cmds-353.100.22).
```

Comment:
I wonder if we should only try this on Linux and on macOS 12.3 or newer?

Reply:
Or macOS 13 or newer. We don't support macOS 12 anyway so I wouldn't worry to much either way.